src/core/thread/osthread.d

```diff
@@ -1415,1204 +965,613 @@ class Thread
         else version (Posix)
             sched_yield();
     }
+}
 
+private Thread toThread(ThreadBase t) @safe nothrow @nogc pure
+{
+    return cast(Thread) t;
```

Comment:
As you don't expect this to fail, going through dynamic casting should be avoided. `cast(Thread) cast(void*) t` should be a noop for optimized code.